test/files/expand.iced

```diff
@@ -0,0 +1,126 @@
+{prng} = require 'crypto'
+{expand_json,stub_json,hmac_obj} = require '../../lib/expand'
+triplesec = require('triplesec')
+pgp_utils = require('pgp-utils')
+{katch,json_stringify_sorted} = pgp_utils.util
+{KeyManager} = require('kbpgp').kb
+{prng} = require 'crypto'
+{constants,errors,alloc,Cryptocurrency} = require '../../'
+{createHash} = require 'crypto'
+{make_ids} = require '../../lib/base'
+{make_esc} = require 'iced-error'
+{errors} = require '../../lib/errors'
+
+new_uid = () -> prng(15).toString('hex') + "19"
+new_username = () -> "u_" + prng(5).toString('hex')
+
+twiddle_hex = (s) ->
+  b = Buffer.from(s, 'hex')
+  b[0] ^= 1
+  return b.toString('hex')
+
+make_expansions = (v) ->
+  ret = {}
+  for obj in v
+    key = prng(16)
+    hmac = hmac_obj({obj, key}).toString('hex')
+    ret[hmac] = { obj, key : key.toString('hex') }
+  return ret
+
+exports.happy = (T,cb) ->
+  expansion_vals = [ ["a","b","c"], 300 ]
+  expansions = make_expansions expansion_vals
+  h = Object.keys(expansions)
+
+  json = {
+    a : 10
+    b : "hi"
+    c : d : e : h[0]
+    f : [1,2,3]
+    g : [
+      { h : h[1] }
+    ]
+    i : h
+  }
+  res = expand_json { json, expansions }
+  json.c.d.e = expansion_vals[0]
+  json.g[0].h = expansion_vals[1]
+  json.i[0] = expansion_vals[0]
+  json.i[1] = expansion_vals[1]
+  T.equal json, res, "worked"
+  cb null
+
+exports.sad1 = (T,cb) ->
+  expansion_vals = [ ["a","b","c"], 300 ]
+  expansions = make_expansions expansion_vals
+  h = Object.keys(expansions)
+
+  tests = [
+    { expansions : { "a" : ["10"] }, json : 10, err : "bad hash" }
+    { expansions : {}, json : 10, err : "hashcheck failure" }
+    { expansions, json : 10, err : "Did not find expansion for" }
+    { expansion : make_expansions(["💩"]), json : 10, err : "non-ASCII" }
+    { expansions : {}, json : 10, err : "hashcheck failure" }
+    { expansions : {}, json : 10, err : "bad hmac key" }
+  ]
+  expansion = expansions[h[0]]
+  tests[1].expansions[h[0]] = { obj : ["a","b"], key : expansion.key }
+  tests[4].expansions[h[0]] = { obj : expansion.obj, key : twiddle_hex(expansion.key) }
+  tests[5].expansions[h[0]] = { obj : expansion.obj, key : expansion.key[0...12] }
+
+  for t in tests
+    try
+      expand_json t
+    catch e
```

Comment:
does this get run if there was no error thrown?

Reply:
good point, and a test doesn't work...

Reply:
fixed